Tools/automation.ps1

```diff
@@ -0,0 +1,791 @@
+        Az.Accounts  --> Connect-AzAccount, Get-AzContext, Set-AzContext
+        Az.KeyVault  --> Set-AzKeyVaultSecret
+        Az.Websites  --> Publish-AzWebapp
+        Az.Functions --> Get-AzFunctionApp
+    
+    .PARAMETER ResourcePrefix
+    Azure resources will be created with the "ResourcePrefix" prefixed to the name.
+    
+    .PARAMETER Index
+    Azure resources will be created with the "Index" sufixed to the name.
+    
+    .PARAMETER AzResourceGroup
+    Azure Resource Group that will be created or used to group the Azure resources being created by this script.
+    
+    .PARAMETER AzSubscriptionName
+    Azure Subscription that will have the Azure Resource Group and Azure resources created in (optional).
+    
+    .PARAMETER AzLocation
+    Azure location that will be used for the created resources. (Default: westus)
+    
+    .PARAMETER WorkingDirectory
+    Local file repository that will be referenced by this script. Used to locate the Template files, and identifies where the Parameters will be created (Default: Script Location).
+    
+    .PARAMETER ShowConnectionInstructions
+    Outputs to the screen after all Azure objects have been created, the command to add the new private repository to your Windows Package Manager client.
+
+    .EXAMPLE
+    .\automation.ps1 -ResourcePrefix "contoso-" -Index "Demo" -AzResourceGroup "WinGet_PrivateRepo_Demo"
+
+    .EXAMPLE
+    .\automation.ps1 -ResourcePrefix "contoso-" -Index "001" -AzResourceGroup "WinGet_PrivateRepo" -AzLocation "westus"
+
+    .EXAMPLE
+    .\automation.ps1 -ResourcePrefix "contoso-" -Index "Prod" -AzResourceGroup "WinGet_PrivateRepo_Prod" -AzSubscription "Contoso Global" -ShowConnectionInstructions
+
+    .NOTES
+    Something
+    #>
+
+param(
+    [Parameter(Position=0, Mandatory=$true)]  [string]$ResourcePrefix,
+    [Parameter(Position=1, Mandatory=$true)]  [string]$Index,
+    [Parameter(Position=2, Mandatory=$true)]  [string]$AzResourceGroup,
+    [Parameter(Position=3, Mandatory=$false)] [string]$AzSubscriptionName,
+    [Parameter(Position=4, Mandatory=$false)] [string]$AzLocation         = "westus",
+    [Parameter(Position=5, Mandatory=$false)] [string]$WorkingDirectory   = "$($(Get-Item $PSScriptRoot).fullname)\..\src\WinGet.RestSource.Infrastructure",
+    [Parameter(Position=6, Mandatory=$false)] [string]$ArchiveFunctionZip = "CompiledFunctions.zip"
+)
+
+Function Test-RequiredModules
+{
+    Param(
+        [Parameter(Position=0, Mandatory=$true)] [string]$RequiredModule
+    )
+    Begin
+    {}
+    Process
+    {
+        ## Determinds if the PowerShell Module is missing, If missing Returns the name of the missing module
+        IF(!$(Get-Module -ListAvailable -Name $RequiredModule) )
+            { $Result = $RequiredModule }
+    }
+    End
+    {
+        ## Returns a value only if the module is missing
+        Return $Result
+    }
+}
+
+Function New-WinGetRepo
```

Comment:
Recommendation: add documentation at top of each function which explains what the function does.

Reply:
Added commented Descriptions to the top of each Function in the Automation.ps1 script.